app/src/main/java/com/quran/labs/androidquran/util/QuranFileUtils.java

```diff
@@ -355,8 +355,21 @@ public static String getQuranAyahDatabaseDirectory(Context context) {
 
   @Nullable
   public static String getQuranAudioDirectory(Context context){
-    String s = QuranFileUtils.getQuranBaseDirectory(context);
-    return (s == null)? null : s + AUDIO_DIRECTORY + File.separator;
+    String path = getQuranBaseDirectory(context);
+    if (path == null) {
+      return null;
+    }
+    path += AUDIO_DIRECTORY;
+    File dir = new File(path);
+    if (dir.exists() && !dir.isDirectory() && !dir.delete()) {
```

Comment:
>on the same note though, just so i am 100% sure - this writes `.nomedia` in `audio/` 

Yes, that's right; only under `audio/`.

> will the various media players know that if there's `.nomedia` in audio, that there doesn't need to be one in `audio/{qari}`?

If the media player is written correctly and gets data from framework's mediastore, then `.nomedia` in the root folder is enough. But I am not sure regarding the old Android versions. I will test on an emulator with older platform later and report here...